plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -345,6 +350,36 @@ func (s *SQLServer) Stop() {
 	}
 }
 
+func (s *SQLServer) gatherServerWithRetry(pool *sql.DB, query Query, acc telegraf.Accumulator, connectionString string) error {
+	for retryAttempt := 0; retryAttempt <= s.RetryCount; retryAttempt++ {
+		queryError := s.gatherServer(pool, query, acc, connectionString)
+		if queryError == nil {
+			// gatherServer completed successfully
+			break
+		}
```

Comment:
Is there a way to tell if the error is recoverable by retry? For example, if the connection is down the error might be recoverable, but if the credentials are wrong or the metric is malformed you will add (useless) traffic (through the retries).